crates/sui-graphql-rpc/schema/current_progress_schema.graphql

```diff
@@ -1741,6 +1754,9 @@ type PageInfo {
 	endCursor: String
 }
 
+"""
+The parent of this object
```

Comment:
Unless we intend for other parent child object relationships, I feel like we can just explain that a child object is a dynamic field on some parent object and so on

Reply:
I have not really understood the relationship yet, it's quite confusing to me. 
So if this object has a parent object, that means that this object is a dynamic field on this parent object? 

I am confused by all these concepts: Parent, Owner, Dynamic Field, Child, Object, as I do not understand the relative relationship between them. 

Happy to put here whatever description would be least confusing and self contained to avoid any confusion.

Reply:
Parent + child relationship only makes sense if you're talking about an object (parent) and its dynamic fields (children.)

Otherwise, Owner == ObjectKind as this indicates whether it is owned by some address (ObjectOwner::Address), is immutable (ObjectOwner::Immutable), shared (ObjectOwner::Shared) or a dynamic field (ObjectOwner::Parent)

So if the confusion stems from the fact that instead of describing the object's kind, we're now talking about the object's owner's type (Address, Immutable, Shared, Parent), we could do something like:

keep it back to kind instead of owner
kind = ObjectKind {
  Owned(AddressOwner),
  Immutable(Immutable),
  Child(ObjectOwner),
  Shared(Shared)
}

query:
kind {
  ... on Owned {
    owner {
      address
    }
  ... on Immutable { _ }
  ... on Child { parent { ... } }
  ... on Shared { initialSharedVersion }

And to address @amnn 's comment 

> I think it's kind of confusing in Rust to call this Parent, but it's way way more confusing in the GraphQL schema to have the struct that represent's an object's parent be called Child.

By keeping 'kind' instead of 'owner', we can now understand this as, the object's kind is Child, and it has parent xyz. @stefan-mysten would that clear your confusion (and likely the same confusion from other graphql-rpc users)